spec/features/course/homepage_spec.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.feature 'Course: Homepage' do
+  let(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    let(:course) { create(:course, :opened) }
+    let(:course_student) { create(:course_student, :approved, course: course) }
+    let!(:valid_announcement) { create(:course_announcement, course: course) }
```

Comment:
I think let! is necessary because I need to visit a page in this test, and if the announcement is lazily evaluated(let), the announcement will only be created after the page is generated, which means the page won't contain the announcement..